subsys/bluetooth/host/conn.c

```diff
@@ -1210,17 +1210,6 @@ int bt_conn_send_cb(struct bt_conn *conn, struct net_buf *buf,
 static bool conn_tx_internal(bt_conn_tx_cb_t cb)
 {
 	if (cb == att_pdu_sent || cb == att_cfm_sent || cb == att_rsp_sent ||
-#if defined(CONFIG_BT_SMP)
-#if defined(CONFIG_BT_PRIVACY)
-	    cb == smp_id_sent ||
-#endif /* CONFIG_BT_PRIVACY */
-#if defined(CONFIG_BT_SIGNING)
-	    cb == smp_sign_info_sent ||
-#endif /* CONFIG_BT_SIGNING */
-#if !defined(CONFIG_BT_SMP_SC_PAIR_ONLY)
-	    cb == smp_ident_sent ||
-#endif /* CONFIG_BT_SMP_SC_PAIR_ONLY */
-#endif /* CONFIG_BT_SMP */
```

Comment:
Would it be wise to add a code comment here to avoid re-introducing a similar bug? Something like "note: SMP functions are not considered internal since they may cause application callbacks". Or perhaps: "Only functions which are known to not block on anything that may depend on the TX thread itself (thereby causing a deadlock) are considered internal in this context. This means that e.g. SMP callbacks cannot be considered internal, since they may perform application callbacks (which we don't control)."

Reply:
That's a good point. Added comment based on your suggestion.